bugbug/model.py

```diff
@@ -156,31 +165,116 @@ def get_human_readable_feature_names(self):
+            # to each feature across the whole dataset.
+            shap_sums = shap_values.sum(0)
+            abs_shap_sums = np.abs(shap_values).sum(0)
+            rel_shap_sums = abs_shap_sums / abs_shap_sums.sum()
+
+            cut_off_value = cutoff * np.amax(rel_shap_sums)
+
+            # Get indices of features that pass the cut off value
+            top_feature_indices = np.where(rel_shap_sums >= cut_off_value)[0]
+            # Get the importance values of the top features from their indices
+            top_features = np.take(rel_shap_sums, top_feature_indices)
+            # Gets the sign of the importance from shap_sums as boolean
+            is_positive = (np.take(shap_sums, top_feature_indices)) >= 0
+            # Stack the importance, indices and shap_sums in a 2D array
+            top_features = np.column_stack(
+                (top_features, top_feature_indices, is_positive)
+            )
+            # Sort the array (in decreasing order of importance values)
+            top_features = top_features[top_features[:, 0].argsort()][::-1]
+
+            return top_features
+
+        important_features = {}
+        important_features["classes"] = {}
+        important_features["average"] = get_top_features(
+            cutoff, np.sum(np.abs(shap_values), axis=0)
+        )
+        for num, item in enumerate(shap_values):
+            # top features for that class
+            top_item_features = get_top_features(cutoff, item)
+
+            # shap values of top average features for that class
+            abs_sums = np.abs(item).sum(0)
+            rel_sums = abs_sums / abs_sums.sum()
+            is_pos = ["+" if shap_sum >= 0 else "-" for shap_sum in item.sum(0)]
+            top_avg = [
+                is_pos[int(index)] + str(rel_sums[int(index)])
+                for importance, index, is_positive in important_features["average"]
+            ]
+
+            important_features["classes"][class_names[num]] = (
+                top_item_features,
+                top_avg,
+            )
 
-        cut_off_value = cutoff * np.amax(rel_shap_sums)
+        return important_features
 
-        # Get indices of features that pass the cut off value
-        top_feature_indices = np.where(rel_shap_sums >= cut_off_value)[0]
-        # Get the importance values of the top features from their indices
-        top_features = np.take(rel_shap_sums, top_feature_indices)
-        # Gets the sign of the importance from shap_sums as boolean
-        is_positive = (np.take(shap_sums, top_feature_indices)) >= 0
-        # Stack the importance, indices and shap_sums in a 2D array
-        top_features = np.column_stack((top_features, top_feature_indices, is_positive))
-        # Sort the array (in decreasing order of importance values)
-        top_features = top_features[top_features[:, 0].argsort()][::-1]
+    def print_feature_importances(
+        self, important_features, feature_names, classify=False
```

Comment:
`classify` is not a great name, and it's not great that it can be either a boolean or a list, it should be consistent.
You could call it `class_probabilities` and make it None in the non-classify case and a list in the classify case.